8bitzz/COSC2101
# Fix/cart item count

### Change-log
- Introduce state `count` with 2 handlers `handleItemAdd` & `handleItemRemove` on App.js
- Use `context.count` for displaying number of cart items on Navigation Bar
- Handle remove item on **Cart** page using `context.handleItemRemove`
- Handle add item on **Detail** page, **Card Hover**, and **Featured** section using `context.handleItemAdd`

### How-to-test
- Checkout this branch and start server for both frontend & backend
- Open `Network` tab in console and verify that `/api/v1/carts` is called in the following events:
- [ ] Called once on Homepage to display number of items in cart 
- [ ] Called if only users add/remove items in their Cart
- [ ] Called if only users go to Cart page

**Filtered Movies Page**
<img width="1675" alt="Screen Shot 2021-08-25 at 17 48 19" src="https://user-images.githubusercontent.com/23531403/130777580-61b7c4c0-ca41-42af-988c-a5a0fd94eee9.png">

**Cart Page**
<img width="1675" alt="Screen Shot 2021-08-25 at 17 48 46" src="https://user-images.githubusercontent.com/23531403/130777632-63961be8-2695-4e9f-9a03-19ce00d16338.png">

**Remove Item on Cart Page**
<img width="1674" alt="Screen Shot 2021-08-25 at 17 49 06" src="https://user-images.githubusercontent.com/23531403/130777664-6dccca16-7f07-42c3-8b58-b59ce5a51694.png">

**Add Item from Homepage**
<img width="1676" alt="Screen Shot 2021-08-25 at 17 50 26" src="https://user-images.githubusercontent.com/23531403/130777881-05efc31b-cfde-4b0d-8dd4-f9dda37ff289.png">
